services/webnn/ort/graph_builder_ort.cc

```diff
@@ -1030,6 +1030,38 @@ void GraphBuilderOrt::AddUnaryOperation(const T& operation,
   model_editor_.AddNode(op_type, node, inputs, outputs);
 }
 
+void GraphBuilderOrt::AddElementWiseSignOperation(
+    const mojom::ElementWiseUnary& sign) {
+  CHECK_EQ(sign.kind, mojom::ElementWiseUnary::Kind::kSign);
+  std::string input = GetOperandNameById(sign.input_operand_id);
+  OperandDataType input_data_type =
+      GetOperand(sign.input_operand_id).descriptor.data_type();
+
+  // OV EP GPU does not support integer data type for sign operation, so insert
+  // a cast operation to convert the sign input to float data type.
+  bool need_cast = false;
+  if (input_data_type == OperandDataType::kInt8 ||
+      input_data_type == OperandDataType::kInt32 ||
+      input_data_type == OperandDataType::kInt64) {
+    input = PrependCast(input, ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT);
```

Comment:
int32 and int64 values can not be fully represented by float. I think we should report integers are not supported by sign operator on OV GPU through opSupportLimits.

Reply:
It's a OV GPU only issue, reporting error through op SupportLimits will also let all ORT contexts with different EPs/Devices fail to run sign with integer input. Probably it's better to report in the sign impl method in graph_builder_ort.cc if
`if (base::FeatureList::IsEnabled(mojom::features::kWebNNOrtOpenVino) && ... device=GPU...)?` WDYT?

Reply:
Or should we get different opSupportlimits according to different EPs/devices?

Reply:
The latter, opSupportLimits should be implementation dependent, e.g. EP/device.

Reply:
OK, I will update.

Reply:
Updated, PTAL, thanks!